pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -69,6 +69,10 @@
   public static final String GROUPBY_TRIM_THRESHOLD = "groupby.trim.threshold";
   public static final int DEFAULT_GROUPBY_TRIM_THRESHOLD = 1_000_000;
 
+  // set as pinot.server.query.executor.use.dictionary.for.distinct;
+  public static final String USE_DICTIONARY_FOR_DISTINCT = "use.dictionary.for.distinct";
```

Comment:
One issue with server side configs is that different servers of the same table may potentially have different settings (due to user error). Perhaps this could be a query option, if we want to start with off-by-default?